test/server/cards/01_SOR/units/GeneralKrellHeartlessTactician.spec.ts

```diff
@@ -98,7 +98,60 @@ describe('General Krell, Heartless Tactician', function() {
                 expect(context.player1.handSize).toBe(startingHandSize + 1);
             });
         });
+        describe('Krell\'s ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['the-emperors-legion'],
+                        groundArena: ['battlefield-marine', 'general-krell#heartless-tactician', 'reputable-hunter']
+                    },
+                    player2: {
+                        groundArena: ['wampa', 'atst'],
+                        hand: ['waylay']
+                    }
+                });
+            });
 
-        // TODO WAYLAY: once we have a card implemented that can return cards from discard to hand, add a test confirming that the ability is regained correctly
+            it('should draw only 1 card when returned to hand and played again.', function() {
+                const { context } = contextRef;
+                const startingHandSize = context.player1.handSize;
+
+                // CASE 1: We defeat krell, return him to hand, play him again and defeat a friendly unit so we draw 1 card
+                context.player1.clickCard(context.generalKrell);
+                context.player1.clickCard(context.atst);
+
+                context.player2.passAction();
+                context.player1.clickCard(context.theEmperorsLegion);
+                context.player2.passAction();
+
+                context.player1.clickCard(context.generalKrell);
+                context.player2.passAction();
+                context.player1.clickCard(context.battlefieldMarine);
+                context.player1.clickCard(context.wampa);
+
+                expect(context.player1).toHavePrompt('Trigger the ability \'Draw a card\' or pass');
+                context.player1.clickPrompt('Draw a card');
+                expect(context.player2).toBeActivePlayer();
+
+                // its the same as the starting hand since we play 1 card
+                expect(context.player1.handSize).toBe(startingHandSize);
+
+                // CASE 2: we Waylay krell and return him to hand and play him again
```

Comment:
For the tests of basic systems (as opposed to card) I think are the tests where I'm okay with having more cases, just because there aren't that many of them and they're so important so we want to know exactly which case failed quickly if something breaks. So for all of the tests in here except Krell I think we can break things out into individual cases a little more than we normally would.